index-handlers/src/main/java/no/sikt/nva/nvi/index/UpdateIndexHandler.java

```diff
@@ -101,16 +103,32 @@ private static boolean isCandidate(DynamodbStreamRecord record) {
         return CANDIDATE_TYPE.equals(extractRecordType(record));
     }
 
+    private static AuthorizedBackendUriRetriever defaultUriRetriver() {
+        return new AuthorizedBackendUriRetriever(new Environment().readEnv("BACKEND_CLIENT_AUTH_URL"),
+                                                 new Environment().readEnv("BACKEND_CLIENT_SECRET_NAME"));
+    }
+
+    private static void logRecordThatCouldNotBeIndexed(DynamodbStreamRecord record) {
+        var eventString = attempt(() -> JsonUtils.dtoObjectMapper.writeValueAsString(record)).orElseThrow();
+        LOGGER.error(COULD_NOT_UPDATE_INDEX_MESSAGE, eventString);
+    }
+
     private boolean isCandidateOrApproval(DynamodbStreamRecord record) {
         return isCandidate(record) || isApproval(record);
     }
 
     private void updateIndex(DynamodbStreamRecord record) {
-        var candidate = nviService.findCandidateById(extractIdentifierFromOldImage(record)).orElseThrow();
-        if (isApplicable(candidate)) {
-            addDocumentToIndex(candidate);
-        } else {
-            removeDocumentFromIndex(candidate);
+        try {
+            var candidateIdentifier = extractIdentifierFromNewImage(record);
+            var candidate = candidateRepository.findCandidateById(candidateIdentifier).orElseThrow();
```

Comment:
Why use `candidateRepository` instead of `CandidateBO.fromRequest()` here?

Reply:
We need periodRepository to fetch CandidateBO, there is no need for it in this case.